torch/_refs/__init__.py

```diff
@@ -2775,22 +2776,25 @@ def stack(tensors: TensorSequenceType, dim: int = 0) -> TensorLikeType:
     return torch.cat([t.unsqueeze(wrapped_dim) for t in tensors], dim)
 
 
+# CompositeImplicitAutograd - don't register decomp
 @out_wrapper()
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
```

Comment:
I don't see why you need a type promotion wrapper here as opposed to previous casting to computation dtype. Besides, it's just wrong - softmax doesn't do int_to_float type promotion, it errors out for integer args.

Reply:
i assumed it would be cleaner this way. `int to float` is due to using `div` (as in `addcdiv`), so we'd get the correct(?) behavior if ints are supported later. _i'll change back to explicit cast to computation dtype tho_

Reply:
pushed a version that can use `dtype` before converting to comp dtype and when converting to result dtype